### PR TITLE
[Enhancement] Support create analyze job for external table

### PR DESCRIPTION
**Pull request by Youngwb**
Fixes #issue
Support create analyze job for externla table
1. do not support sample analyze job, only support full analyze
2. do not support analyze all database
3. do not support analyze all tables in one db
4. only support analyze specific table

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java [hunk 1]
```diff
@@ -130,13 +130,33 @@ public Void visitCreateAnalyzeJobStatement(CreateAnalyzeJobStmt statement, Conne
                 if ((Strings.isNullOrEmpty(tbl.getCatalog()) &&
                         CatalogMgr.isExternalCatalog(session.getCurrentCatalog())) ||
                         CatalogMgr.isExternalCatalog(tbl.getCatalog())) {
-                    throw new SemanticException("External Table don't support analyze job");
+                    if (tbl.getTbl() == null) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java:133]: it's better to normalizationTableName first

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/CreateAnalyzeJobStmt.java [hunk 2]
```diff
@@ -17,21 +17,22 @@
...
+    private String catalogName;
     private long dbId;
     private long tableId;
     private final TableName tbl;
     private final List<String> columnNames;
     private final boolean isSample;
     private Map<String, String> properties;
-
-    private final boolean isNative;
+    private boolean isNative;
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/sql/ast/CreateAnalyzeJobStmt.java:35]: USE `catalogName` to check isNative ?

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java (hunk 1 above)
Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java:133]: We can normalize the tableName, because it's tbl could be null.

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 166 / 197 (84.26%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/StatisticsCollectJob.java|0|1|00.00%|[86]|
|:large_blue_circle:|com/starrocks/qe/DDLStmtExecutor.java|0|4|00.00%|[778, 779, 780, 782]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|0|1|00.00%|[2593]|
|:large_blue_circle:|com/starrocks/statistic/StatisticAutoCollector.java|2|13|15.38%|[108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsCollectJobFactory.java|48|60|80.00%|[119, 128, 138, 147, 156, 163, 196, 201, 202, 203, 204, 205]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java|20|22|90.91%|[150, 151]|
|:large_blue_circle:|com/starrocks/statistic/ExternalAnalyzeJob.java|66|66|100.00%|[]|
|:large_blue_circle:|com/starrocks/statistic/AnalyzeMgr.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/statistic/ExternalFullStatisticsCollectJob.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateAnalyzeJobStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/persist/gson/GsonUtils.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowAnalyzeJobStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|8|8|100.00%|[]|
|:large_blue_circle:|com/starrocks/journal/JournalEntity.java|6|6|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by stephen-shelby — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.2

**Pull request merged**